File: Java/lab7/src/main/java/org/example/Realisation/PcOrderAdapter.java
package org.example.Realisation;

import org.example.Interface.IPhoneOrder;
public class PcOrderAdapter implements IPhoneOrder {
    private PcOrder pcOrder;

    public PcOrderAdapter(PcOrder pcOrder) {
        this.pcOrder = pcOrder;
    }
    @Override
    public void createPhoneOrder() {
        pcOrder.createPcOrder();
    }
}

File: Python/lab7/Realisation/PcOrderAdapter.py
from Interfaces.IPhoneOrder import IPhoneOrder
from Realisation.PcOrder import PcOrder

class PcOrderAdapter(IPhoneOrder):
    def __init__(self, pc_order: PcOrder):
        self.pc_order = pc_order

    def createPhoneOrder(self):
        self.pc_order.createPcOrder()
